### PR TITLE
Made improvements to scale up and down for streams and consumers.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/consumer.go [hunk 1]
```diff
@@ -749,6 +749,16 @@ func (o *consumer) checkQueueInterest() {
...
 }
 
+// clears our node if we have one. When we scale down to 1.
+func (o *consumer) clearNode() {
+	o.mu.Lock()
+	if o.node != nil {
+		o.node.Delete()
+		o.node = nil
+	}
+	defer o.mu.Unlock()
```
Comment (kozlovic) [inline server/consumer.go:759]: Weird to have the defer here. Either you have this after lock, or remove defer.

Comment (kozlovic) [reply server/consumer.go:759]: Note also that although that we often check for o.node != nil, I did not find that we were setting it to nil, except now.

File: server/stream.go [hunk 2]
```diff
@@ -3241,6 +3240,11 @@ func (mset *stream) internalLoop() {
 			c.flushClients(0)
 			outq.recycle(&pms)
 		case <-msgs.ch:
+			// This can possible change now so needs to be checked here.
```
Comment (kozlovic) [inline server/stream.go:3243]: Typo? "This can possible change"

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: server/consumer.go (hunk 1 above)
Comment (derekcollison) [reply server/consumer.go:759]: Will fix.

We set it in setConsumerAssignment.

**Review by kozlovic — APPROVED** [review]
lgtm

**Pull request merged**